controlplane/internal/controller/agentclusterinstall_controller.go

```diff
@@ -79,6 +82,16 @@ func (r *AgentClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 			log.Error(err, "failed setting OpenshiftAssistedControlPlane to ready")
 			return ctrl.Result{}, err
 		}
+		//TODO: determine a better condition to add this
+		kubeConfigSecret, err := r.getACIKubeconfig(ctx, aci, acp)
+		if err != nil {
+			log.Error(err, "failed to get Kubeconfig secret for workload client")
+			return ctrl.Result{}, err
+		}
+		if err := setControlPlaneVersion(ctx, kubeConfigSecret, &acp); err != nil {
```

Comment:
If we just set the ready condition true of "WorkloadKubeconfig available" to the OACP, we could then move this logic to the controlplane controller and trigger it by the condition being ready. We could also remove the code retrieving the kubeconfig as it's already done in the reconcile function - we can retrieve it from the well known secret for kubeconfig (our secret, not the ACI's) - WDYT?

Reply:
Great idea! I'll implement that since retrieving the kubeconfig here is kind of clunky and the condition I have for setting that version is unideal too.